src/features/schema/fields/location/LocationBased.jsx

```diff
@@ -1,160 +1,387 @@
-    }
+        // Call handler with new location to update location-dependent options
+        callHandler(field.id, field.handler, getLocationAsJson(newValue));
+    };
 
+    // Handler for dropdown option selection
     const onChangeOption = (event) => {
         let newValue = { ...value };
         newValue.value = event.target.value;
+
+        // Get display text from selected option
         const selectedOption = options.find(option => option.value === event.target.value);
         if (selectedOption) {
             newValue.display = selectedOption.display;
         }
+
+        // Update state and Redux store
         setValue(newValue);
         dispatch(set({
             id: field.id,
             value: JSON.stringify(newValue),
         }));
-    }
+    };
 
+    // Get options from handler results for the dropdown
     let options = [];
     if (field.id in handlerResults.values) {
         options = handlerResults.values[field.id];
     }
 
     return (
         <FormControl fullWidth>
-            <Typography>Latitude</Typography>
-            <InputSlider
-                min={-90}
-                max={90}
-                step={0.1}
-                onChange={onChangeLatitude}
-                value={value['lat']}
-            >
-            </InputSlider>
-            <Typography>Longitude</Typography>
-            <InputSlider
-                min={-180}
-                max={180}
-                step={0.1}
-                onChange={onChangeLongitude}
-                value={value['lng']}
-            >
-            </InputSlider>
-            <Typography>Locality</Typography>
-            <TextField
-                fullWidth
-                variant="outlined"
-                onChange={onChangeLocality}
-                style={{ marginBottom: '0.5rem' }}
-                value={value['locality']}
+            {/* Location search field */}
+            <Typography variant="subtitle1" gutterBottom>Location</Typography>
+            <LocationPicker
+                onLocationSelect={handleLocationSelect}
+                initialLocation={value}
             />
-            <Typography>Timezone</Typography>
-            <Select
-                onChange={onChangeTimezone}
-                style={{ marginBottom: '0.5rem' }}
-                value={value['timezone']}
-            >
-                {Intl.supportedValuesOf('timeZone').map((zone) => {
-                    return <MenuItem value={zone}>{zone}</MenuItem>
-                })}
-            </Select>
-            <Typography>Options for chosen location</Typography>
+
+            {/* Display selected location details */}
+            {value && (
+                <div className="mt-4 p-3 border rounded bg-gray-50">
+                    <Typography variant="body2"><strong>Selected:</strong> {value.locality}</Typography>
+                    <Typography variant="body2"><strong>Coordinates:</strong> {value.lat.toFixed(3)}, {value.lng.toFixed(3)}</Typography>
```

Comment:
I get `[Error] TypeError: s.lat.toFixed is not a function. (In 's.lat.toFixed(3)', 's.lat.toFixed' is undefined)` on this line.

Reply:
is this when typing in specific coordinates to the picker ?